pkg/webhook/webhook.go

```diff
@@ -174,6 +192,68 @@ func (m *podMutator) mutateContainers(containers []corev1.Container, clientID st
 	return containers
 }
 
+func (m *podMutator) injectProxyInitContainer(containers []corev1.Container, proxyPort int32) []corev1.Container {
+	shouldInject := true
+	for _, container := range containers {
+		if strings.HasPrefix(container.Image, ProxyInitImageRepository) || container.Name == ProxyInitContainerName {
+			shouldInject = false
+			break
```

Comment:
nit: could we just return here instead of break and get rid of `shouldInject` bool var altogether?